Tasks pagination

BMartinos opened the pull request:
This PR relates to issue: https://github.com/jembi/openhim-console/issues/255

Added filtering and pagination additions to the getTask() and getTasks() API function. Updated tests. Found a bug in src/tasks.coffee that broke the server

BMartinos commented:
Hey @devcritter Do you mind reviewing this PR for me when you get a chance. Thanks

devcritter commented on src/api/tasks.coffee:
```diff
@@ -128,12 +158,71 @@ exports.getTask = (taskId) ->
+    tempTransactions = result.transactions
+
+    # are filters present
+    if Object.keys( filters ).length > 0
```
This method's getting a little big and it might be a good idea to split it up into smaller units of work. This section looks like a good candidate. This could easily be split into a method `buildFilteredTransactionsArray` or something similar.

devcritter commented on test/integration/tasksAPITests.coffee:
```diff
@@ -326,8 +342,23 @@ describe "API Integration Tests", ->
     describe '*getTask(taskId)', ->
```
It might be good to add some more tests for getTask: two examples I can think of are:
* Add a task with failed and completed transactions, and see if the filter only returns the failed transactions
* Add a task with >10 transactions and check that the returned number is 10 (with a limit turned specified)

devcritter commented:
Thanks @BMartinos! It looks great. Just some comments above.

BMartinos commented:
Hey @devcritter I have added some more test coverage for the filtering and moved the transaction array filtering to a function. Mind just reviewing this for me again. Thanks

devcritter commented:
:+1: Thanks that looks great
